api/src/wfl/module/covid.clj

```diff
@@ -25,8 +18,65 @@
 (defn create-covid-workload!
   [tx {:keys [source sink executor] :as request}])
 
-(defn start-covid-workload!
-  [])
+;; TODO: move this to the queue-based skeleton later
+(defn ^:private snapshot-created?
+  "Check if `snapshot-job-id` has done in TDR,
+   return the resulting snapshot id if so."
+  [snapshot-job-id]
+  (some->> snapshot-job-id
+           (datarepo/job-done?)
+           :id))
+
+(defn ^:private go-create-snapshot!
+  "Create snapshot in TDR from `dataset`, `table`
+   and `row-ids` and write job info as well as rows
+   into `source-details-name` table."
+  [tx dataset table source-details-name [start end] row-ids]
+  (let [columns     (-> (datarepo/all-columns dataset table)
+                        (->> (map :name) set)
+                        (conj "datarepo_row_id"))
+        job-id (-> (datarepo/make-snapshot-request dataset columns table row-ids)
+                   (update :name util/randomize)
+                   ;; FIXME: this sometimes runs into: Cannot update iam permissions, need failure handling here!
+                   (datarepo/create-snapshot))]
+    (jdbc/insert! tx
+                  source-details-name
+                  {:snapshot_creation_job_id job-id
+                   :datarepo_row_ids row-ids
+                   :start_time start
+                   :end_time end})))
+
+(defn ^:private check-for-new-data!
+  "Check for new data in TDR, create new snapshots, insert
+   resulting job creation ids into database and update the
+   timestamp for next time using transaction TX."
+  [tx
+   {:keys [source_item] :as workload}
+   {:keys [dataset table last_checked snapshot details] :as _source}]
+  (let [now (util/utc-now)
+        dataset (datarepo/dataset dataset)
```

Comment:
Perhaps lift this out of the function? This information won't change so we probably only need to store it once - see comment about caching above?